activesupport/lib/active_support/evented_file_update_checker.rb

```diff
@@ -86,6 +90,10 @@ def initialize(files, dirs)
         @mutex = Mutex.new
 
         start
+        # inotify / FSEvents file descriptors are inherited on fork, so
+        # we need to reopen them otherwise only the parent or the child
+        # will be notified.
+        # FIXME: this callback is keeping a reference on the instance
         @after_fork = ActiveSupport::ForkTracker.after_fork { start }
```

Comment:
Interesting. So closing the FD on Linux is what break it? That's weird.

Reply:
@byroot Yeah.. no idea :joy: Thanks for taking a look!